force-app/main/default/classes/Homework.cls

```diff
@@ -63,38 +82,50 @@ public with sharing class Homework {
     {
         // Declare two Integers, with the values 6 and 7, in a single 
         // statement.
+        Integer firstValue = 6, secondValue = 7;
     }
 
     public void booleanFromString()
     {
         // Create a boolean whose value is initialised from the text
         // ‘true’.
+        
+        Boolean myBooleanValue = Boolean.valueOf('true');
+
     }
 
     public void reverse()
     {
         // Create a string with the value ‘1234567890’ 
-        
+        String leftToRight = '1234567890';
         // Use a built-in method to convert to ‘0987654321’
+        String rightToLeft = leftToRight.reverse();
     }
 
     public void findApex(String candidate) 
     {
         // Find the position of the substring “Apex” in candidate
+        //Bhargava is learning Apex programming
+        String apex = 'Apex';
+        Integer substringPosition = 'Bhargava is learning Apex programming'.indexOf(apex);
+
```

Comment:
This is close, but you seem to be ignoring the parameter for these challenges. The comment says to find the substring apex in candidate, which is a parameter. You should be using that rather than creating a new string literal.